### DISCUSSION TITLE
Failure to compress a model

### DISCUSSION THREAD

**Discussion by XTFan07** [deepmodeling/deepmd-kit#787] [Q&A]
I use 'module load deepmd/compress dp compress input.json -i 1.3-model.pb -o compressed-model.pb' to compress a model and it goes wrong. How can I adress this problem?

Traceback (most recent call last):
File "/data/share/apps/deepmd/compress/bin/dp", line 8, in
sys.exit(main())
File "/data/share/apps/deepmd/compress/lib/python3.8/site-packages/deepmd/main.py", line 327, in main
compress(**dict_args)
File "/data/share/apps/deepmd/compress/lib/python3.8/site-packages/deepmd/entrypoints/compress.py", line 108, in compress
freeze(checkpoint_folder=checkpoint_folder, output=output, node_names=None)
File "/data/share/apps/deepmd/compress/lib/python3.8/site-packages/deepmd/entrypoints/freeze.py", line 147, in freeze
input_graph_def = graph.as_graph_def()
File "/data/share/apps/deepmd/compress/lib/python3.8/site-packages/tensorflow/python/framework/ops.py", line 3294, in as_graph_def
result, _ = self._as_graph_def(from_version, add_shapes)
File "/data/share/apps/deepmd/compress/lib/python3.8/site-packages/tensorflow/python/framework/ops.py", line 3211, in _as_graph_def
graph.ParseFromString(compat.as_bytes(data))
google.protobuf.message.DecodeError: Error parsing message

**Comment by njzjz** [comment]
Please report the version you are using

**Reply by XTFan07** [reply]
Thanks for your reply. The version I use is a test branch with compression function, and the same error was also found when using v2.0.0.beta1.

**Reply by felix5572** [reply]
@XTFan07  take a look? The reason may be that the compressed model is too large #790

**Reply by XTFan07** [reply]
Thank you! Do you mean that I can use larger "decay_steps"  value to make the compressed model smaller?

**Reply by felix5572** [reply]
see: https://github.com/deepmodeling/deepmd-kit/blob/devel/doc/getting-started.md#compress-a-model

the `step` option in `dp compress` command

**Reply by XTFan07** [reply]
OK. Thanks so much. I will have a try.

**Reply by XTFan07** [reply]
It works! Thanks!

**Reply by XTFan07** [reply]
Although the model has been successfully compressed, the MD simulation using the compressed model gives incorrect structures. Is this due to the lower accuracy of the smaller model size? How to set the step value more reasonably?

**Reply by felix5572** [reply]
Yes, try and error for now... In my experience. The MD trace will be nearly the same for at least the first 100 steps.(energy, pressure, temperature.the difference is within 1e-4 )

You could compare the `lammps.log` file with compressed and uncompressed model. (shouble be  the same for the first frame in MD.).

**Reply by XTFan07** [reply]
OK. Thanks!

**Comment by njzjz** [comment]
This error cannot be fixed due to tensorflow limitations. But I'm adding a message in #839.
